### DISCUSSION TITLE
SafeList for 'after' don't work

### DISCUSSION THREAD

**Discussion by chlbri** [tailwindlabs/tailwindcss#9300] [Help]
**What version of Tailwind CSS are you using?**
3.1.8

**What build tool (or framework if it abstracts the build tool) are you using?**
SolidJS

**What version of Node.js are you using?**
16

**What browser are you using?**
edge

**What operating system are you using?**
macOS,

**Describe your issue**

`/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ['./src/**/*.{js,jsx,ts,tsx}'],
  theme: {
    extend: {},
  },
  plugins: [],
  safelist: [
    'after:absolute',
    'after:rounded-full',
    'after:w-2.5',
    'after:h-2.5',
    'after:bg-cyan-600',
    {
      pattern: /(rounded-full|bg-cyan-600|absolute|w-2.5|h-2.5|ml-1)/,
      variants: ['after'],
    },
  ],
};`

My tailwind config

I create a classList inside a hook in solid like that :
` const classList: main.ClassList = {
    'after:rounded-full  after:bg-cyan-600 after:absolute after:w-2.5 after:h-2.5 after:ml-1':
      dragging(),
  };
`

But it doesn't work. The after component is not drawind

**Comment by adamwathan** [comment]
Can you provide a full reproduction? The first thing that stands out to me is you are missing a display type like `after:block`, without that nothing will render.

**Comment by chlbri** [comment]
Yes, it works. Because I'm working with solidjs, so things are not updated like react. I need to create a getter . Something like this : 

`
const _dragClass = 'after:rounded-full after:bg-cyan-600 after:absolute after:w-2.5 after:h-2.5 after:ml-1'

return {
...
  get classList() {
    return {
      [_dragClass]: dragging(),
    };
  }
}`
